IdentitySdkCore/IdentitySdkCore/Classes/ReachFiveMfa.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+import BrightFutures
+import Alamofire
+
+public enum StartMfaCredentialRegistrationRequest {
+    case Email(redirectUrl: String?)
+    case PhoneNumber(phoneNumber: String)
+}
+
+public enum VerifyMfaCredentialRegistrationRequest {
+    case Email(verificationCode: String)
+    case PhoneNumber(verificationCode: String, phoneNumber: String)
+}
+
+public extension ReachFive {
+    
+    func addMfaCredentialRegistrationCallback(mfaCredentialRegistrationCallback: @escaping MfaCredentialRegistrationCallback) {
+        self.mfaCredentialRegistrationCallback = mfaCredentialRegistrationCallback
+    }
+    
+    func startMfaCredentialRegistration(authToken: AuthToken, request: StartMfaCredentialRegistrationRequest) -> Future<MfaStartCredentialRegistrationResponse, ReachFiveError> {
```

Comment:
Le nommage des méthodes et paramètres est très différent en swift que dans un langage de type Java/Swift.
Les labels d'argument font parti intégrante du nom de la méthode.

Cf. [Function Argument Labels and Parameter Names](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/functions/#Function-Argument-Labels-and-Parameter-Names)

Par exemple pour cette méthode,  XCode donne son nom comme : 
`startMfaCredentialRegistration(authToken:request:)`

Avec cela vient le fait que l'écriture des labels est obligatoire lors de l'appel de la méthode, et récursivement si les arguments sont des objets complexe.
On se retrouve avec des choses comme ça qui ne sont pas très agréable à lire : 
```swift
startMfaCredentialRegistration(
    authToken: authToken,
    request: StartMfaCredentialRegistrationRequest.Email(redirectUrl: nil)
)
```

Je proposerais plutôt quelque chose comme : 
```swift
startMfa(registering: .Email(), authToken: authToken)
```

Le type de crédential, qui est la chose la plus importante passe en premier argument.
Le mot crédential est utilisé comme paramètre et comme nom de l'enum au lieu de `StartMfaCredentialRegistrationRequest`.
Aussi généralement on ne met pas le nom de l'enum quand on les utilise. 
```swift
public enum Credential {
    case Email(redirectUrl: String? = nil)
    case PhoneNumber(_ phoneNumber: String)
}
```
J'ai mis une valeur par défaut à l'argument dans Email et omis le label pour le paramètre `phoneNumber`.

Aussi ce serait bien que l'on commence à écrire des Doc pour nos méthodes d'API. J'ai un peu commencé à le faire, ici ça pourrait donner quelque chose comme ça :

```swift
/// Start the registration of a `credential` for use in Multi-Factor Authentification
func startMfa(registering credential: Credential, authToken: AuthToken) -> ...
```

Si tu veux tu peux jeter un œil à cette doc qui donne des conseils pour créer des API

https://www.swift.org/documentation/api-design-guidelines/

Reply:
Toutes les méthodes de l'API commenceraient par `mfa`

Reply:
Comme dit dans Slack je n'ai pas encore un avis définitif sur la question.
Pour l'instant je penche sur `mfaStart(registering:authToken:)` et `mfaVerify(credential:code:authToken:)` mais pour faire un choix définitif il faudrait concevoir toutes les méthodes MFA ensemble